cscs-checks/apps/greasy/greasy_check.py

```diff
@@ -0,0 +1,218 @@
+
+        self.tags = {'production'}
+
+    @rfm.run_before('run')
+    def generate_tasks_file(self):
+        with open(os.path.join(self.stagedir, self.tasks_file), 'w') as outfile:
+            for i in range(self.num_greasy_tasks):
+                outfile.write("./%s output-%d\n" % (self.executable, i))
+
+    @rfm.run_before('run')
+    def daint_dom_gpu_specific_workaround(self):
+        if self.current_partition.fullname in ['daint:gpu', 'dom:gpu']:
+            self.variables['CRAY_CUDA_MPS'] = "1"
+            self.variables['CUDA_VISIBLE_DEVICES'] = "0"
+            self.variables['GPU_DEVICE_ORDINAL'] = "0"
+
+            self.extra_resources = {
+                'gres': {
+                    'gres': 'gpu:0,craynetwork:4'
+                }
+            }
+
+    @rfm.run_before('run')
+    def change_executable_name(self):
+        # After compiling the code we can change the executable to be
+        # the greasy one
+        self.executable = 'greasy'
+
+    @rfm.run_before('run')
+    def set_launcher(self):
+        # The job launcher has to be changed to local since greasy
+        # make calls to srun
+        self.job.launcher = getlauncher('local')()
+
+    @sn.sanity_function
+    def eval_sanity(self):
+        output_files = []
+        for file in os.listdir(self.stagedir):
+            if file.startswith("output-"):
+                output_files.append(file)
+
+        num_greasy_tasks = len(output_files)
+        failure_msg = ('Requested %s task(s), but only executed %s tasks(s)' %
+                       (self.num_greasy_tasks, num_greasy_tasks))
+        sn.evaluate(sn.assert_eq(num_greasy_tasks, self.num_greasy_tasks,
+                                 msg=failure_msg))
+
+        num_tasks = sn.getattr(self, 'nranks_per_worker')
+        num_cpus_per_task = sn.getattr(self, 'num_cpus_per_task')
+
+        def tid(match):
+            return int(match.group(1))
+
+        def num_threads(match):
+            return int(match.group(2))
+
+        def rank(match):
+            return int(match.group(3))
+
+        def num_ranks(match):
+            return int(match.group(4))
+
+        for output_file in output_files:
+            result = sn.findall(r'Hello, World from thread \s*(\d+) out '
+                                r'of \s*(\d+) from process \s*(\d+) out of '
+                                r'\s*(\d+)', output_file)
+
+            failure_msg = ('Found %s Hello, World... pattern(s), but expected '
+                           '%s pattern(s) inside the output file %s' % (
+                                              sn.count(result),
+                                              num_tasks * num_cpus_per_task,
+                                              output_file))
+            sn.evaluate(sn.assert_eq(sn.count(result),
+                                     num_tasks * num_cpus_per_task,
+                                     msg=failure_msg))
+
+            sn.evaluate(sn.assert_true(sn.all(
```

Comment:
Why do you need the `sn.assert_true` here?

Reply:
nope. you are right